app/misc/git.ts

```diff
@@ -495,6 +495,56 @@ function createBranch() {
   }
 }
 
+// search for tags
+function searchTag() {
+  Git.Repository.open(repoFullPath)
+    .then(function (repo) {
+      repo.getCurrentBranch()
+        .then(function () {
+          // grab the list of references - these could be branches or tags
+          return repo.getReferences(Git.Reference.TYPE.LISTALL);
+        }).then(function (refList) {
+          refList.sort();
+          bname = {};
+          tags = {};
+          for (let i = 0; i < refList.length; i++) {
+
+            // strip name for readability
+            let refName = refList[i].name().split("/")[refList[i].name().split("/").length - 1];
+
+            if (refList[i].isTag()){
+              if (refName != document.getElementById("tag-name").value) {
+                document.getElementById(refName).setAttribute("style","display:none");
+              } else {
+                document.getElementById(refName).setAttribute("style","display:block");
+              }
+            }
+
+            lastRefList = refList.slice(); // update lastRefList
+        })
+    });
+}
+
+function displayTag(name, id, onclick) {
```

Comment:
This function already exists in repo.ts?

Reply:
Yes - I thought repo.ts is only called when the repo is initialized. If this isn't the case, then I can just remove the duplicate function.